exec/java-exec/src/main/java/org/apache/drill/exec/planner/sql/handlers/SqlHandlerConfig.java

```diff
@@ -42,6 +43,7 @@ public class SqlHandlerConfig {
 
   private final QueryContext context;
   private final SqlConverter converter;
+  private ConcurrentHashMap<RelNode, Collection<StoragePlugin>> scannedPluginCache = new ConcurrentHashMap<>();
```

Comment:
I think having a regular hash map should be enough, there shouldn't be concurrent queries for this class, it is created per query.

And please use the interface for the reference type.